virtinst/guest.py

```diff
@@ -862,6 +862,22 @@ def _convert_to_vnc_graphics(self):
             dev.add_child(listen)
         dev.set_defaults(self)
 
+    def _convert_to_vnc_video(self):
+        """
+        If there's no video device, add a default one.
+        If there's any qxl device, reset its config to app defaults.
+        """
+        if not self.devices.video:
+            videodev = DeviceVideo(self.conn)
+            videodev.set_defaults(self)
+            self.add_device(videodev)
+            return
+
+        qxl_devs = [v for v in self.devices.video if v.model == "qxl"]
+        for dev in qxl_devs:
+            dev.clear()
+            dev.set_defaults(self)
```

Comment:
This will fail for some VMs, for example Windows 11 will get `bocsh`. Libvirt allows only `qxl` or `virtio` as secondary video device, any other type is rejected.

We need to hard-code `virtio` here or use
``` python
        if (guest.lookup_domcaps().supports_video_virtio() and
            guest.osinfo.supports_virtiogpu()):
```
to decide if we convert it to `virtio` or remove it.

Reply:
Another nice catch :) Ok I fixed it without the open coding check. Just call `dev.set_defaults()` and if we get back a non-virtio model for a non-primary device, remove the device.